src/main/java/br/com/caelum/vraptor/gson/LocalDateGsonConverter.java

```diff
@@ -0,0 +1,63 @@
+/***
+ * Copyright (c) 2009 Caelum - www.caelum.com.br/opensource
+ * All rights reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.caelum.vraptor.gson;
+
+import com.google.gson.*;
+
+import javax.enterprise.context.Dependent;
+import java.lang.reflect.Type;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Date;
+
+/**
+ * A gson converter for {@link LocalDate}.
+ *
+ * @author Ivo Sestren Junior
+ */
+@Dependent
+public class LocalDateGsonConverter implements JsonDeserializer<LocalDate>, JsonSerializer<LocalDate> {
+
+    private final SimpleDateFormat iso8601Format;
+
+    public LocalDateGsonConverter() {
+        this.iso8601Format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
+    }
+
+    @Override
+    public LocalDate deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
+        try {
+            Date date = iso8601Format.parse(json.getAsString());
+
+            return LocalDate.from(date.toInstant());
```

Comment:
@IvoSestren, cant we use `LocalDate[Time]#parse` instead?

https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html#parse-java.lang.CharSequence-java.time.format.DateTimeFormatter-

what do you think?